Formula/c/clang-uml.rb

```diff
@@ -0,0 +1,123 @@
+class ClangUml < Formula
+  desc "Customizable automatic UML diagram generator for C++ based on Clang"
+  homepage "https://github.com/bkryza/clang-uml"
+  url "https://github.com/bkryza/clang-uml/archive/refs/tags/0.5.3.tar.gz"
+  sha256 "e830363ec510f14cc738c6509107b3f52bc55ececc2e27c068cadb093604e943"
+  license "Apache-2.0"
+  head "https://github.com/bkryza/clang-uml.git", branch: "master"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "pkg-config" => :build
+  depends_on "llvm"
+  depends_on "yaml-cpp"
+
+  fails_with gcc: "5"
+
+  def llvm
+    deps.map(&:to_formula)
+        .find { |f| f.name.match?(/^llvm(@\d+)?$/) }
+  end
+
+  def install
+    args = %W[
+      -DBUILD_TESTS=OFF
+      -DCLANG_UML_ENABLE_BACKTRACE=OFF
+      -DCMAKE_INSTALL_RPATH=#{rpath(source: libexec/"bin", target: llvm.opt_lib)}
+    ]
+
+    # If '.git' directory is not available during build, we need
+    # to provide the version using a CMake option
+    args.append("-DGIT_VERSION=#{version}") unless build.head?
+
+    # clang-uml needs to be linked to LLVM c++ when compiled with Clang, otherwise
+    # Trace/BPT trap is raised whenever an exception is thrown (even if it's handled properly)
+    if OS.mac?
+      cc_bin = llvm.opt_bin/"clang"
+      cxx_bin = llvm.opt_bin/"clang++"
+      args.append(%Q(-DCMAKE_EXE_LINKER_FLAGS="-L#{llvm.opt_lib}/c++ -Wl,-rpath,#{llvm.opt_lib}/c++"))
+    end
+
+    with_env(CC: cc_bin, CXX: cxx_bin) do
+      system "cmake", "-S", ".", "-B", "build", *std_cmake_args(install_prefix: libexec), *args
+    end
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    bin.write_exec_script libexec.glob("bin/*")
```

Comment:
If we're going to symlink everything into `bin` anyway, why do we need to install into `libexec`?

Reply:
@carlocab This is a 'hack' I copied from this formula https://github.com/Homebrew/homebrew-core/blob/4e9350530b0aa46024d8b874feadaba338a9c796/Formula/i/include-what-you-use.rb - it didn't help, so once I resolve the issue on `macos13,14/arm64` I'll change it...